packages/salesforcedx-vscode-apex/src/commands/forceLaunchReplayDebugger.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  EmptyParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from '@salesforce/salesforcedx-utils-vscode/out/src';
+import {
+  Command,
+  CommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import {
+  notificationService
+} from '@salesforce/salesforcedx-utils-vscode/out/src/commands';
+import * as vscode from 'vscode';
+import { nls } from '../messages';
+import {
+  testOutlineProvider
+} from '../views/testOutlineProvider';
+import {
+  forceAnonApexDebug
+} from './forceApexExecute';
+
+export async function forceLaunchReplayDebugger() {
+  const editor = vscode.window.activeTextEditor;
+  if (!editor) {
+    notificationService.showErrorMessage(
+      nls.localize('unable_to_locate_editor')
+    );
+    return;
+  }
+
+  const sourceUri = editor.document.uri;
+  if (!sourceUri) {
+    notificationService.showErrorMessage(
+      nls.localize('unable_to_locate_document')
+    );
+    return;
+  }
+
+  if (fileIsAnonymousApex(sourceUri)) {
```

Comment:
Does this mean debugger only works for .apex and not for any file that has anonymous apex selected text? (Sorry I'm late to this. Maybe this was already discussed and resolved?)

Reply:
That is correct.  It has two modes/paths which it runs on:
A) it's filename ends with .apex
B) it's an Apex file, and is a, Apex test file (verified by testOutlineProvider.getTestClassName())

Reply:
Update: it now also handles .log files.